backend/src/stwo/prover.rs

```diff
@@ -68,11 +169,47 @@ where
                 .circle_domain()
                 .half_coset,
         );
+        //TODO: make machines with multi degree sizes work, one only the first one is taken, multi degrees error is handled by NoVariableDegreeAvailable in mod.rs
+        let (trees, mut prover_channel) = self
+            .proving_key
+            .borrow_mut()
+            .as_mut()
+            .and_then(|stark_proving_key| stark_proving_key.preprocessed.values_mut().next())
+            .and_then(|table_collection| table_collection.values_mut().next())
+            .map(|table_proving_key| {
+                (
+                    std::mem::take(&mut table_proving_key.trees),
+                    std::mem::take(&mut table_proving_key.prover_channel),
```

Comment:
`RefCell` and` std::mem::take ` are removed. Proving_key now stores only the constant_trace evaluations in circle domain. 
I tried to store commitment_scheme in proving_key, it is defined as: 
`pub struct CommitmentSchemeProver<'a, B: BackendForChannel<MC>, MC: MerkleChannel> {
    pub trees: TreeVec<CommitmentTreeProver<B, MC>>,
    pub config: PcsConfig,
    twiddles: &'a TwiddleTree<B>,
}`
the field twiddles needs to specify 'a, it propagate to all the functions calling it, I couldn't make it work. I then store the field trees in proving_key only, it then has the problem of ownership, it doesn't implement clone trait. 
so now I store the constant_trace evaluations in proving_key, and do the commit in prove function.  
constant_trace still needs to be cloned to commit due to ownership, I didn't manage to avoid it. Is there any other way to avoid this clone?

Reply:
I see. Let's keep it like this for now.